haystack/modeling/model/multimodal/transformers.py

```diff
@@ -0,0 +1,256 @@
+    """
+
+    @silence_transformers_logs
+    def __init__(
+        self,
+        pretrained_model_name_or_path: Union[str, Path],
+        model_type: str,
+        content_type: str,  # replace with ContentTypes starting Python3.8
+        model_kwargs: Optional[Dict[str, Any]] = None,
+        feature_extractor_kwargs: Optional[Dict[str, Any]] = None,
+        pooler_kwargs: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        :param pretrained_model_name_or_path: The name of the model to load.
+        :param model_type: the value of `model_type` from the model's `Config` class
+        :param content_type: The type of data (such as "text", "image", and so on) the model should process.
+            See the values of `haystack.schema.ContentTypes`.
+        :param model_kwargs: A dictionary of parameters to pass to the model's initialization (`revision`, `use_auth_key` and the like)
+            Haystack applies some default parameters to some models. You can ovewrite them by specifying the
+            desired value in this parameter. See `DEFAULT_MODEL_PARAMS`.
+        :param feature_extractor_kwargs: A dictionary of parameters to pass to the feature extractor's initialization (`revision`, `use_auth_key` and the like)
+            Haystack applies some default parameters to some models. You can overwrite them by specifying the
+            desired value in this parameter. See `DEFAULT_EXTRACTION_PARAMS`.
+        :param pooler_kwargs: A dictionary of parameters to pass to the pooler's initialization (`summary_last_dropout`, `summary_activation`, and so on)
+            Haystack applies some default parameters to some models. You can overwrite them by specifying the
+            desired value in this parameter. See `POOLER_PARAMETERS`.
+        """
+        super().__init__(
+            pretrained_model_name_or_path=pretrained_model_name_or_path,
+            model_type=model_type,
+            content_type=content_type,
+        )
+        logger.debug("Model type: %s", model_type)
+        model_class: PreTrainedModel = getattr(transformers, model_type + "Model", None)
+        logger.debug("Model class: %s", model_class)
+        self.model = model_class.from_pretrained(str(pretrained_model_name_or_path), **(model_kwargs or {}))
+
+        # Create a feature extractor
+        feature_extractor_kwargs = {"do_lower_case": True, **(feature_extractor_kwargs or {})}
+        self.feature_extractor = FeatureExtractor(
+            pretrained_model_name_or_path=pretrained_model_name_or_path, **feature_extractor_kwargs
+        )
+
+        # The models with an entry in POOLER_PARAMETERS do not provide a pooled_output by default.
+        if POOLER_PARAMETERS.get(self.model_type, None) is not None or pooler_kwargs is not None:
+
+            # FIXME: We used to not have a dropout in the end of the pooler, because it was done in the prediction head.
+            #   Double-check if we need to add it here.
```

Comment:
No, thanks!

So, I checked and indeed it seems we should add a dropout layer here: regular Bert models don't return meaningful results right now. I've had a look at the prediction heads but I can't seem to find there is the relevant code :sweat_smile: If you have an idea please point me to it, otherwise I'm going to dig it out.